src/app/components/plugin-uiframe/plugin-uiframe.component.ts

```diff
@@ -297,14 +296,18 @@ export class PluginUiframeComponent implements OnChanges, OnDestroy {
     }
 
     private loadImplementations(): void {
+        const experimentId = this.parent.experimentId;
```

Comment:
These changes (making the workspace injectable and not using the currentExperimentServce to get the experimentId) seem unrelated. Why are these changes required?